File: CS/ValidationErrorHostBehavior/ViewModels/MainViewModel.cs
using DevExpress.Mvvm;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace ValidationErrorHostBehavior.ViewModels
{
    public class MainViewModel : ViewModelBase, IDataErrorInfo
    {
        protected string firstname_;
        [Required]
        [RegularExpression(@"^[a-zA-Z''-'\s]{1,40}$", 
         ErrorMessage = "Characters are not allowed.")]
        public string FirstName
        {
            get { return this.firstname_; }
            set { this.SetProperty(ref this.firstname_, value, "FirstName"); }
        }


        protected string lastname_;
        [Required]
        [RegularExpression(@"^[a-zA-Z''-'\s]{1,40}$", 
         ErrorMessage = "Characters are not allowed.")]
        public string LastName
        {
            get { return this.lastname_; }
            set { this.SetProperty(ref this.lastname_, value, "LastName"); }
        }

        public string Error
        {
            get
            {
                return string.Join(";", this["FirstName"], this["LastName"]);
            }
        }

        public string this[string columnName]
        {
            get
            {
                return IDataErrorInfoHelper.GetErrorText(this, columnName);
            }
        }
    }
}

File: VB/ValidationErrorHostBehavior/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations

Namespace ValidationErrorHostBehavior.ViewModels

    Public Class MainViewModel
        Inherits ViewModelBase
        Implements IDataErrorInfo

        Protected firstname_ As String

        <Required>
        <RegularExpression("^[a-zA-Z''-'\s]{1,40}$", ErrorMessage:="Characters are not allowed.")>
        Public Property FirstName As String
            Get
                Return firstname_
            End Get

            Set(ByVal value As String)
                SetProperty(firstname_, value, "FirstName")
            End Set
        End Property

        Protected lastname_ As String

        <Required>
        <RegularExpression("^[a-zA-Z''-'\s]{1,40}$", ErrorMessage:="Characters are not allowed.")>
        Public Property LastName As String
            Get
                Return lastname_
            End Get

            Set(ByVal value As String)
                SetProperty(lastname_, value, "LastName")
            End Set
        End Property

        Public ReadOnly Property [Error] As String Implements IDataErrorInfo.[Error]
            Get
                Return String.Join(";", Me("FirstName"), Me("LastName"))
            End Get
        End Property

        Default Public ReadOnly Property Item(ByVal columnName As String) As String Implements IDataErrorInfo.Item
            Get
                Return IDataErrorInfoHelper.GetErrorText(Me, columnName)
            End Get
        End Property
    End Class
End Namespace
